.travis.yml

```diff
@@ -29,14 +25,26 @@ before_install:
   - pip install --upgrade git+git://github.com/SpiNNakerManchester/PACMAN.git@$TRAVIS_BRANCH || pip install --upgrade git+git://github.com/SpiNNakerManchester/PACMAN.git@master
   - pip install --upgrade git+git://github.com/SpiNNakerManchester/DataSpecification.git@$TRAVIS_BRANCH || pip install --upgrade git+git://github.com/SpiNNakerManchester/DataSpecification.git@master
   - pip install --upgrade git+git://github.com/SpiNNakerManchester/spalloc.git@${TRAVIS_BRANCH} || pip install --upgrade git+git://github.com/SpiNNakerManchester/spalloc.git@master
-  - pip install --upgrade git+git://github.com/SpiNNakerManchester/SpiNNFrontEndCommon.git@$TRAVIS_BRANCH || pip install --upgrade git+git://github.com/SpiNNakerManchester/SpiNNFrontEndCommon.git@master
+  - FECVER=`git ls-remote https://github.com/SpiNNakerManchester/SpiNNFrontEndCommon.git | grep -q refs/heads/$TRAVIS_BRANCH && echo $TRAVIS_BRANCH || echo master`
+  - pip install --upgrade git+git://github.com/SpiNNakerManchester/SpiNNFrontEndCommon.git@$FECVER
+  # C dependencies
+  - git clone --branch v3.1.1 https://github.com/SpiNNakerManchester/spinnaker_tools.git
```

Comment:
This should also use either the current branch or the master branch to allow synchronous branches.

Reply:
I'm planning to leave spinnaker_tools pinned to 3.1.1 for now; that repository follows its own versioning schedule.